services/NewsService.py

```diff
@@ -39,33 +45,37 @@ def getNews(self):
         )
         news_data = list(news_data)
 
-        news = self.llm(
-            [
-                SystemMessage(
-                    content="You are a news analyzer that will read the given cyber security news and return the answer based on the user's requirement."
-                ),
-                HumanMessage(content=str(news_data)),
-                AIMessage(
-                    content="I have read the news you provided. Now please tell me your request."
-                ),
-                HumanMessage(
-                    content="""Please give me a list of the most recent cybersecurity news based on the news given above.
-                    Indicate the source and date.
-                    Each news should be strictly in the format of {news_format}.
-                    Your reply should be news-only, without adding any of your conversational content.
-                    Do not use bullet points.
-                    Do not stop generating the answer until it is complete.
-                    Return at most {news_number} news.""".format(
-                        news_format=self.news_format, news_number=self.news_number
-                    )
-                ),
-            ]
-        )
+        template = """Question: {question}
+        Answer: Let's think step by step."""
 
-        print(news.content)
+        prompt = PromptTemplate.from_template(template)
+
+        messages = [
+            {
```

Comment:
we can separate these part into a function, like `enrichMessage()`, to improve the code structure.

Reply:
since prompt for keywords and without keywords is different, i will have to make 2 functions. That is not improving the code structure to a great extent. should I stick with the old way?

Reply:
One function can handle these two use cases for sure, we only need to add a condition judgement. 

Actually, we can merge both `getNews` and `getNewsWithKeywords` into a single function. Most of the content in these two functions are the same, except the prompts. We can put the prompts into a separate file which is solely for config/constant content storage. 

It always a good idea to reduce the redundant code by improving the usability of our code. It's just my suggestion, but I highly recommend to take some time to think about how to do this. Even though our code is not complex for now, but in the future, it may not.🤔

Reply:
alright, i will make the changes